omod/src/test/java/org/openmrs/web/controller/user/UserFormControllerTest.java

```diff
@@ -31,18 +34,39 @@
 	private UserFormController controller;
 	
 	/**
-	 * @see UserFormController#handleSubmission(WebRequest,HttpSession,String,String,String,null,User,BindingResult)
-	 * 
+	 * @see UserFormController#handleSubmission(WebRequest,HttpSession,String,String,String,null, String, User,BindingResult)
+	 *
 	 */
 	@Test
-	@Verifies(value = "should work for an example", method = "handleSubmission(WebRequest,HttpSession,String,String,String,null,User,BindingResult)")
+	@Verifies(value = "should work for an example", method = "handleSubmission(WebRequest,HttpSession,String,String,String,null, String, User,BindingResult)")
 	public void handleSubmission_shouldWorkForAnExample() throws Exception {
 		WebRequest request = new ServletWebRequest(new MockHttpServletRequest());
 		User user = controller.formBackingObject(request, null);
 		user.addName(new PersonName("This", "is", "Test"));
 		user.getPerson().setGender("F");
 		controller.handleSubmission(request, new MockHttpSession(), new ModelMap(), "", "Save User", "pass123", "pass123", null,
-		    null, null, new String[0], "true", user, new BindException(user, "user"));
+				null, null, new String[0], "true", null, user, new BindException(user, "user"));
+	}
+	
+	/**
+	 * @see UserFormController#handleSubmission(WebRequest,HttpSession,String,String,String,null, String, User,BindingResult)
+	 *
+	 */
+	@Test
+	@Verifies(value = "Adds ", method = "handleSubmission(WebRequest,HttpSession,String,String,String,null, String, User,BindingResult)")
+	public void handleSubmission_createProviderAccountForUserWhenProviderTableCheckBoxIsSelected() throws Exception {
+		WebRequest request = new ServletWebRequest(new MockHttpServletRequest());
+		User user = controller.formBackingObject(request, null);
+		user.addName(new PersonName("This", "is", "Test"));
+		user.getPerson().setGender("F");
+		user.getPerson().setId(1234);
+		Assert.assertTrue(Context.getProviderService().getProvidersByPerson(user.getPerson()).isEmpty());
+		ModelMap model = new ModelMap();
+		model.addAttribute("isProvider", false);
+		controller.showForm(1234, "true", user, model);
+		controller.handleSubmission(request, new MockHttpSession(), new ModelMap(), "", "Save User", "pass123", "pass123", null,
+				null, null, new String[0] , "true", "addToProviderTable", user, new BindException(user, "user"));
+		Assert.assertNotNull(Context.getProviderService().getProvidersByPerson(user.getPerson()));
```

Comment:
This should be checking that the list is not empty, ProviderService.getProvidersByPerson can never return null